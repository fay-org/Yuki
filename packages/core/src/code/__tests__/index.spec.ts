import Code from '../index'
import { mount } from '@vue/test-utils'

const mockWindowQuery = () => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn()
    }))
  })
}

describe('Code', () => {
  it('should be render as element', () => {
    mockWindowQuery()
    const wrapper = mount(Code)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should support props block', () => {
    mockWindowQuery()
    const wrapper = mount(Code, {
      props: {
        block: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('pre')).toBeTruthy()
  })
})
