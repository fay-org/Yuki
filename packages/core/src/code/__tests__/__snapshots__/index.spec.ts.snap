// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code should be render as element 1`] = `"<code><!----></code>"`;

exports[`Code should support props block 1`] = `
"<div class="fect-code" style="--code-border: #fafafa; --code-bg-color: rgba(250, 250, 250, 0.75); --code-radius: 6px; --code-name-border: #eaeaea; --code-name-bg-color: #eaeaea; --code-name-color: #666; --code-font-size: calc(0.875 * 16px); --code-width: initial; --code-height: auto; --code-pt: calc(1.1 * 16px); --code-pr: calc(1 * 16px); --code-pb: calc(1.1 * 16px); --code-pl: calc(1 * 16px); --code-mt: calc(1.3 * 16px); --code-mr: 0; --code-mb: calc(1.3 * 16px); --code-ml: 0; --code-name-pt: calc(0.32 * 16px); --code-name-pr: calc(0.5 * 16px); --code-name-pb: calc(0.32 * 16px); --code-name-pl: calc(0.5 * 16px); --code-name-font-size: calc(0.8125 * 16px);">
  <!----><pre><!----></pre>
</div>"
`;
